[Core] Add unit tests for LMCacheEngineBuilder.destroy() method

ikaadil opened the pull request:
FILL IN THE PR DESCRIPTION HERE

FIX #xxxx (*link existing issues this PR will resolve*)

**PLEASE READ THE CHECKLIST BELOW AND FILL IN THE DESCRIPTION ABOVE**

---

<details>
<!-- inside this <details> section, markdown rendering does not work, so we use raw html here. -->
<summary><b> PR Checklist (Click to Expand) </b></summary>

<p>Thank you for your contribution to LMCache! Before submitting the pull request, please ensure the PR meets the following criteria. This helps us maintain the code quality and improve the efficiency of the review process.</p>

<h3>PR Title and Classification</h3>
<p>Please try to classify PRs for easy understanding of the type of changes. The PR title is prefixed appropriately to indicate the type of change. Please use one of the following:</p>
<ul>
    <li><code>[Bugfix]</code> for bug fixes.</li>
    <li><code>[CI/Build]</code> for build or continuous integration improvements.</li>
    <li><code>[Doc]</code> for documentation fixes and improvements.</li>
    <li><code>[Model]</code> for adding a new model or improving an existing model. Model name should appear in the title.</li>
    <li><code>[Core]</code> for changes in the core LMCache logic (e.g., <code>LMCacheEngine</code>, <code>Backend</code> etc.)</li>
    <li><code>[Misc]</code> for PRs that do not fit the above categories. Please use this sparingly.</li>
</ul>
<p><strong>Note:</strong> If the PR spans more than one category, please include all relevant prefixes.</p>

<h3>Code Quality</h3>

<p>The PR need to meet the following code quality standards:</p>

<ul>
    <li>The code need to be well-documented to ensure future contributors can easily understand the code.</li>
    <li> Please include sufficient unit tests to ensure the change is stay correct and robust. The unit and integration tests will always run and our comprehensive test will be triggered after the "full" label is tagged onto a PR.</li>
</ul>

<h3>What to Expect for the Reviews</h3>

We aim to address all PRs in a timely manner. If no one reviews your PR within 5 days, please @-mention one of KuntaiDu, ApostaC or YaoJiayi.

</details>

ikaadil committed:
Remove TODO comment for unit test in LMCacheEngineBuilder.destroy method

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ikaadil committed:
Add unit tests for LMCacheEngineBuilder's destroy method

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ApostaC reviewed (approved):
LGTM! Thanks for your contribution!

ApostaC commented:
Please fix the code quality checks before merging.

Reference: https://docs.lmcache.ai/developer_guide/contributing.html#coding-style

ikaadil committed:
Merge branch 'dev' into test/cache-engine-builder-destroy

ikaadil committed:
[Core] Add unit tests for LMCacheEngineBuilder.destroy() method

- Implement comprehensive unit tests for destroy() method
- Test basic destroy functionality and cleanup
- Test destroying non-existent instances (graceful handling)
- Test destroying one instance doesn't affect others
- Verify all internal state cleanup (_instances, _cfgs, _metadatas, _stat_loggers)
- Remove completed TODO comment

Note: Local mypy failure on macOS due to pre-existing MAP_POPULATE platform
compatibility issue in gds_backend.py (unrelated to cache engine changes)

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ikaadil committed:
Merge branch 'dev' into test/cache-engine-builder-destroy

ikaadil committed:
Merge branch 'dev' into test/cache-engine-builder-destroy

ikaadil commented:
> Please fix the code quality checks before merging.
> 
> Reference: https://docs.lmcache.ai/developer_guide/contributing.html#coding-style

Fixed

sammshen reviewed (approved):
LGTM, thanks @ikaadil
